modules/container-converter/assets/js/editor/migrator.js

```diff
@@ -0,0 +1,88 @@
+import sectionMaps from './maps/section';
+import columnMaps from './maps/column';
+
+export default class Migrator {
+	/**
+	 * Migrate element settings into new settings object, using a map object.
+	 *
+	 * @param {Object} settings - Settings to migrate.
+	 * @param {Object} map - Mapping object.
+	 *
+	 * @return {Object}
+	 */
+	static migrate( settings, map ) {
+		return Object.fromEntries( Object.entries( { ...settings } ).map( ( [ key, value ] ) => {
+			const mapped = map[ key ];
+
+			// Remove setting.
+			if ( null === mapped ) {
+				return null;
+			}
+
+			// Simple key conversion:
+			// { old_setting: 'new_setting' }
+			if ( 'string' === typeof mapped ) {
+				return [ mapped, value ];
+			}
+
+			// Advanced conversion using a callback:
+			// { old_setting: ( { key, value, settings } ) => [ 'new_setting', value ] }
+			if ( 'function' === typeof mapped ) {
+				return mapped( { key, value, settings } );
+			}
+
+			return [ key, value ];
+		} ).filter( Boolean ) );
+	}
+
+	/**
+	 * Determine if an element can be converted to a Container.
+	 *
+	 * @param {String} elType
+	 *
+	 * @return {boolean}
+	 */
+	static canConvertToContainer( elType ) {
+		return [ 'section', 'column' ].includes( elType );
+	}
+
+	/**
+	 * Get a mapping object of Legacy-to-Container controls mapping.
+	 *
+	 * @return {Object}
+	 */
+	static getLegacyControlsMapping( elType ) {
+		switch ( elType ) {
+			case 'section':
+				return sectionMaps();
+
+			case 'column':
+				return columnMaps();
+
+			default:
+				return {};
+		}
+	}
+
+	/**
+	 * Normalize element settings (adding defaults, etc.) by elType,
+	 *
+	 * @param {string} elType - Element type.
+	 * @param {Object} settings - Settings object after migration.
+	 *
+	 * @return {Object} - normalized settings.
+	 */
+	static normalizeSettings( elType, settings ) {
+		switch ( elType ) {
+			case 'section':
+				return {
+					...settings,
+					flex_align_items: settings.flex_align_items || 'center',
+					flex_direction: 'row', // Force it to be row.
+				};
+
+			default:
+				return settings;
+		}
+	}
+}
```

Comment:
Maybe try that all those functions use the same config, then it will be easier to add new elType:
```js
static config = {
	section: {
		legacyMapping: sectionMaps(), // is it must be a function, if it depends on some config wait for this config
		normalizeSettings: (settings) => {...settings}
	},
	column: {
		legacyMapping: columnMaps(),
		normalizeSettings: (settings) => {...settings}
	}
}
```

Another option is to create a class for each of them, but maybe it goes too far.

Reply:
Good idea, done